test/functional/treesitter/parser_spec.lua

```diff
@@ -547,6 +547,36 @@ int x = INT_MAX;
     end)
   end)
 
+  describe('when setting the node for an injection', function()
+    before_each(function()
+      insert([[
+print()
+      ]])
+    end)
+
+    it('should not error when encountering optional captures #23100', function()
```

Comment:
```suggestion
    it('handles optional captures #23100', function()
```

or "supports". this test name will already be absurdly long because of all the nesting